index.src.html

```diff
@@ -185,17 +205,43 @@
     When invoked, run these steps:
     1. If |config| is not a <a>valid AudioDecoderConfig</a>, throw a
         {{TypeError}}.
-    2. Run the <a>Configure Decoder</a> algorithm with |config|.
+    2. If {{AudioDecoder/state}} is `"closed"`, throw an {{InvalidStateError}}.
+    3. If the user agent cannot provide a codec implementation to support
+        |config|, throw a {{NotSupportedError}}.
+    4. Set {{AudioDecoder/state}} to `"configured"`.
+    5. <a>Queue a control message</a> to configure the decoder with |config|.
+    6. <a>Run the control message processing loop</a>.
+
+    <a>Running a control message</a> to configure the decoder means running
+    these steps:
+    1. Assign {{AudioDecoder/[[codec implementation]]}} with an implementation
+        supporting |config|.
   </dd>
 
   <dt><dfn method for=AudioDecoder>decode(chunk)</dfn></dt>
   <dd>
     <a>Enqueues a control message</a> to decode the given |chunk|.
 
     When invoked, run these steps:
-    1. Let |output algorithm| be the <a>AudioFrame Output</a> algorithm.
-    2. Run the <a>Decode Chunk</a> algorithm with |chunk| and
-        |output algorithm|.
+    1. If {{AudioDecoder/state}} is not `"configured"`, throw an
+        {{InvalidStateError}}.
+    2. Increment {{AudioDecoder/decodeQueueSize}}.
+    3. <a>Queue a control message</a> to decode the |chunk|.
+    4. <a>Run the control message processing loop</a>.
+
+    Running a control message to decode the chunk means performing these steps:
+    1. Decrement {{AudioDecoder/decodeQueueSize}}
+    2. Let |codec implementation queue| be the result of starting a new
+        <a>parallel queue</a>.
```

Comment:
A new parallel queue each time break sequentially, two decode operation that should be resolved in order won't have this guarantee.

Reply:
I thought the sequence of the two decode operations was preserved by the higher level control message queue? In my model, the control message processing loop doesn't dequeue the next control message until these parallel steps are finished. Is that wrong? Is there an easy remedy?